app/services/exporter.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class Exporter
+  attr_accessor :metadata, :archive_path
+
+  PERSON_FIELDS_TO_EXPORT = %i[
+    name
+    city
+    phone
+    skype
+    linkedin
+    primary_tech
+    english
+    day_of_birth
+    email
+    skills
+    current_position
+    github
+    personal_email
+  ].freeze
+
+  def perform(start_time: nil, end_time: nil)
+    collect_data(start_time, end_time) && archive_data
+  end
+
+  def collect_data(start_time, end_time)
+    @metadata_fields_types = {
+      person: PERSON_FIELDS_TO_EXPORT.map { |attr| { name: attr, type: Person.columns_hash[attr.to_s].type } }
+    }
+
+    attachments = Attachment.where('name ILIKE ?', '%cv%')
+
+    if start_time.present? && end_time.present?
+      Rails.logger.debug("[EXPORT] Collecting data for time range: #{start_time}..#{end_time}")
+      attachments = attachments.where(created_at: start_time..end_time)
+    else
+      Rails.logger.debug("[EXPORT] Collecting all data (time range not specified)")
+    end
+
+    @metadata = attachments.includes(:person).order(:created_at).map do |a|
+      {
+        id: a.file.id,
+        name: a.file_filename.gsub(' ', '_'),
+        content_type: a.file_content_type,
+        path: "#{a.file.backend.directory}/#{a.file.id}",
+        metadata: {
+          person: PERSON_FIELDS_TO_EXPORT.each_with_object({}) { |attr, res| res[attr] = a.person.send(attr.to_s) }
+        }
+      }
+    end
+
+    @metadata.present?
+  end
+
+  def archive_data
+    time_range =
+      if start_time.present? && end_time.present?
+        "#{start_time}_#{end_time}"
+      else
+        'all'
```

Comment:
might add current date as end so you could know when the export was ran

Reply:
all archives have a timestamp (e.g. hrms_data_all_20191205175239.zip in this case)

Reply:
oh, my bad, this is awesome